perfmetrics/scripts/continuous_test/gcp_ubuntu/e2e_tests/build.sh

```diff
@@ -22,11 +22,11 @@ readonly RUN_E2E_TESTS_ON_INSTALLED_PACKAGE=true
 cd "${KOKORO_ARTIFACTS_DIR}/github/gcsfuse"
 echo "Building and installing gcsfuse..."
 # Get the latest commitId of yesterday in the log file. Build gcsfuse and run
-commitId=$(git log origin/$BRANCH_NAME --before='yesterday 23:59:59' --max-count=1 --pretty=%H)
+commitId=$(git log --before='yesterday 23:59:59' --max-count=1 --pretty=%H)
 ./perfmetrics/scripts/build_and_install_gcsfuse.sh $commitId
 
-# To execute tests for a specific branch, ensure you've checked out from that branch first.
-git checkout origin/$BRANCH_NAME
+# To execute tests for a specific commitId, ensure you've checked out from that commitId first.
+git checkout $commitId
```

Comment:
For Reviewer: I'm introducing this change to address a potential issue where we might build a package based on yesterday's commit ID, but then run tests against today's commit ID which may contain breaking changes. This mismatch could lead to test failures.